docs/source/query/overview.rst

```diff
@@ -5,6 +5,57 @@ Querying Snuba
 This guide drives you through the process of authoring and testing a
 Snuba query.
 
+
+Understanding the cost of queries
+=================================
+The cost of a snuba query has many factors but these are key to keep in mind when constructing queries:
+
+* Amount of rows scanned
+* Amount of columns scanned
+* Type of data scanned
+
+Let's take this query as an example::
+
+    MATCH events
+    SELECT count() AS count
+    WHERE 
+        timestamp > "2021-01-01" AND 
+        timestamp <= "2021-01-02" AND
+        project_id = 1
+ 
+The query above will scan two columns: timestamp and project id over one day. Timestamps and integers are also easy comparisons for a computer to make (can be done in one instruction). Thus this query is fairly cheap. The only way to make this query cheaper is to reduce the time window and scan fewer rows.
+
+Now let's see how we can make this query more expensive.
+
+Expanding the time frame::
+
+    MATCH events
+    SELECT count() AS count
+    WHERE 
+        timestamp > "2021-01-01" AND 
+        timestamp <= "2021-03-01" AND
+        project_id = 1
+ 
+
+Adding more fields to scan. Any extra fields that are scanned are extra files which have to be opened by clickhouse::
+
+    MATCH events
+    SELECT count() AS count
+    WHERE 
+        timestamp > "2021-01-01" AND 
+        timestamp <= "2021-03-01" AND
+        project_id = 1 AND
+        browser = "firefox" AND
+        os = "ios" AND
+        url = "https://santry.io/some_path" AND
+        tags[duration_group] = IN tuple("<10s", "<30s")
```

Comment:
This would be a reasonable usage of tags. The query should become:
`has(_tags_hashmap, cityHash64("duration_group:<10s")) OR has(_tags_hashmap, cityHash64("duration_group:<30s"))`. Which uses the index. It is ok but it gets worse if all rows have that specific tag with that specific value so the index does not exclude anything.

See my message below for details on how to deal with nested columns. This is a more complicated matter.

Reply:
> has(_tags_hashmap, cityHash64("duration_group:<10s")) OR has(_tags_hashmap, cityHash64("duration_group:<30s"))

Are you sure it would become that? Is there a query processor that expands `IN` tag to `OR`s?

Reply:
would it become like that if you provide an `IN`. No
That query should be expressed (as of today) as a series of OR conditions. 
We could write a processor to optimize that
The problem for the reader though is that I would interpret the paragraph and the example as "conditions on tags are bad" while it is more nuanced: 
- equality conditions on tags are ok
- IN can be expressed as a series of equalities in OR
- if your tag has extremely low selectivity the result is still bad. as the index does not exclude anything.